libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -42,12 +44,11 @@ class MapboxDirectionsSession(
         routeOptions: RouteOptions,
         routesRequestCallback: RoutesRequestCallback?
     ) {
-        routes = emptyList()
```

Comment:
That's what we need to confirm i.e. if removing this line everything works 👌 
This line is here because when doing a request we clean up the list because the request could fail. One thing that we can do though is make sure it's clean up when an error occurs. Would that work? cc @LukasPaczos

Reply:
> One thing that we can do though is make sure it's clean up when an error occurs (and remove this line of code). Would that work?

I think it would. An alternative would be exposing a mechanism similar to the faster route, that would work in parallel to the main route request and wouldn't clear the route when the request is started, but only when it fails.